.github/workflows/build-image.yml

```diff
@@ -0,0 +1,98 @@
+name: Build New Image Version
+on:
+  # Manually call
+  workflow_dispatch:
+    inputs:
+      release-type:
+        required: true
+        type: choice
+        description: Type of release
+        options:
+          - patch
+          - minor
+          - major
+      base-version:
+        required: true
+        description: Base version=
+  # Call from other workflow
+  workflow_call:
+    inputs:
+      release-type:
+        type: string
+        required: true
+      base-version:
+        type: string
+        required: true
+defaults:
+  run:
+    shell: bash -l {0}
+jobs:
+  open-pr:
+    name: Open Pull Request
+    runs-on: ubuntu-latest
+    if: github.repository == 'aws/sagemaker-distribution'
+    permissions:
+      pull-requests: write
+      contents: write
+    outputs:
+      pr_id: ${{ steps.open_pr.outputs.pr_id }}
+      target_version: ${{ steps.calc_target.outputs.target_version }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-file: ./environment.yml
+          environment-name: sagemaker-distribution
+          init-shell: bash
+      - name: Free up disk space
+        run: rm -rf /opt/hostedtoolcache
+      - name: Activate sagemaker-distribution
+        run: micromamba activate sagemaker-distribution
+      - name: Calculate target version
+        id: calc_target
+        run: |
+          TARGET_VERSION=$(python -c 'import semver; print(semver.bump_${{ inputs.release-type }}("${{ inputs.base-version }}"))')
+          echo "target_version=$TARGET_VERSION" >> $GITHUB_OUTPUT
+      - name: Create new branch
+        run: git checkout -b release-${{ steps.calc_target.outputs.target_version }}
+      - name: Generate artifacts
+        run: python ./src/main.py create-${{ inputs.release-type }}-version-artifacts --base-patch-version ${{ inputs.base-version }}
```

Comment:
A note about contributions process as per https://github.com/aws/sagemaker-distribution/blob/main/CONTRIBUTING.md#for-adding-new-python-packages-to-sagemaker-distribution

We have a gap where only changes that are package additions can be pre-contributed to an upcoming release. Because of how this create-version-artifacts script works, I don't yet see a way that Dockerfile or initialization script changes can be made. Til now, those types of changes have always been manually applied prior to opening release PRS. What's our plan to support this with the automation?

Reply:
So Dockerfile and other non-env.in changes would be made in templates/v*/Dockerfile. And then the only changes made to env files would be adding packages, as any version changes would be done automatically.